luaui/Include/blueprint_substitution/definitions.lua

```diff
@@ -0,0 +1,164 @@
+-- luaui/Include/blueprint_substitution/definitions.lua
+-- Contains unit category definitions for blueprint substitution
+-- Used by logic.lua
+
+-- Make a global table for these definitions
+WG.BlueprintDefinitions = WG.BlueprintDefinitions or {}
+local BpDefs = WG.BlueprintDefinitions -- Local alias for convenience
+
+-- ===================================================================
+-- Configuration & Data
+-- ===================================================================
+Spring.Log("BlueprintDefs", LOG.INFO, "Initializing blueprint category definitions...")
+
+-- Define sides directly here since we're in unsynced code and can't access gamedata/sidedata.lua
```

Comment:
You can actually access `gamedata/sidedata.lua` here (either via `Spring.GetSideData` or `VFS.Include`) but I'm not sure how useful it would be.

Reply:
Yeah originally I had started to modify that file to have an actual completion/match my own opinion that enums should be inclusive and represent real values, but then I noped out. Right now that seems to be a data structure driving the side selection options. It wasn't that hard maybe one day.

Reply:
I think just do the refactor? The more I stare at this the more it makes sense just to refactor it?

Reply:
Do we like Sides here? I was unsure what ubiquitous language was here. It's firmly sides in code but factions in the UI. We could take this opportunity to rename sides to factions if we're touching sidedata?